### PR TITLE
Add retry mechanism to avoid failures for some steps 

### PR DESCRIPTION
**Pull request by wangke19**
To avoid failures, add retry mechanism for some steps. Such as:
```
oc config refresh-ca-bundle
Maybe hit the 'error: failed to update CA bundle: using system CA bundle to verify server, not allowing refresh to overwrite', 

oc --kubeconfig="${SHARED_DIR}/admin.kubeconfig" whoami
Maybe hit the 'error: You must be logged in to the server (Unauthorized)'
```

### CODE REVIEW

**Comment by wangke19** [comment]
/pj-rehearse rehearse-40789-periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-aws-ipi-byo-iam-role-cert-rotation-p2-f14

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-vsphere-ipi-proxy-fips-cert-rotation-p2-f14

**Comment by wangke19** [comment]
/pj-rehearse  periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.10-aws-ipi-ovn-fips-cert-rotation-p2-f1

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-vsphere-ipi-proxy-fips-cert-rotation-p2-f14

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-vsphere-ipi-proxy-fips-cert-rotation-p2-f14

File: ci-operator/step-registry/openshift/upgrade/qe/test/cert-rotation/openshift-upgrade-qe-test-cert-rotation-commands.sh [hunk 1]
```diff
@@ -126,6 +126,8 @@ oc adm wait-for-stable-cluster
 
 # create new admin.kubeconfig
 oc config new-admin-kubeconfig > "${SHARED_DIR}/admin.kubeconfig"
+oc --kubeconfig="${SHARED_DIR}/admin.kubeconfig" whoami || true
```
Comment (liangxia) [inline ci-operator/step-registry/openshift/upgrade/qe/test/cert-rotation/openshift-upgrade-qe-test-cert-rotation-commands.sh:129]: Looks like this line is not needed, as we have the check after the sleep ?

Comment (wangke19) [reply ci-operator/step-registry/openshift/upgrade/qe/test/cert-rotation/openshift-upgrade-qe-test-cert-rotation-commands.sh:129]: we need to confirm if there is something wrong with this step after retry and the new kubeocnfig works fine.

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.10-azure-aro-cert-rotation-p1-f1

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-gcp-ipi-workload-identity-rt-fips-cert-rotation-p2-f1

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-gcp-ipi-workload-identity-rt-fips-cert-rotation-p2-f1

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-stable-4.12-azure-ipi-fips-cert-rotation-p2-f14

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-vsphere-ipi-proxy-fips-cert-rotation-p2-f14

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-vsphere-ipi-proxy-fips-cert-rotation-p2-f14

**Comment by wangke19** [comment]
Code change has been tested in another PR https://github.com/openshift/release/pull/40995.
Test results:https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/40995/rehearse-40995-periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.10-azure-ipi-fips-cert-rotation-p2-f1/1676804125293023232/artifacts/azure-ipi-fips-cert-rotation-p2-f1/openshift-upgrade-qe-test-cert-rotation/build-log.txt
```
All clusteroperators are stable
+ oc config refresh-ca-bundle
CA bundle for cluster "ci-op-m3612qrq-fa64b" updated.
+ :
...
+ oc --kubeconfig=/tmp/secret/admin.kubeconfig whoami
error: You must be logged in to the server (Unauthorized)
+ sleep 2
+ oc --kubeconfig=/tmp/secret/admin.kubeconfig whoami
system:admin
```

**Comment by xingxingxia** [comment]
1. I hit `'error: You must be logged in to the server (Unauthorized)'` in manual test and retrying worked, so the corresponding improvement of it LGTM.
2. I did not hit `'error: failed to update CA bundle: using system CA bundle to verify server, not allowing refresh to overwrite'` in manual test and not yet know more about it. But adding LGTM anyway since Ke knows more about it.


/lgtm
One minor question is the `sleep 2` and `3` hard code seconds always enough?

**Pull request merged**